.github/workflows/test-xamarinandroid.yml

```diff
@@ -53,57 +61,71 @@ jobs:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
       with:
-        api-level: 29
-        arch: x86_64
-        force-avd-creation: false
-        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
-        disable-animations: false
-        script: echo "Generated AVD snapshot for caching."
+        aws-access-key-id: ${{ secrets.AWS_DEVICEFARM_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_DEVICEFARM_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2
     - name: Run the tests
-      uses: reactivecircus/android-emulator-runner@v2
+      uses: realm/aws-devicefarm/test-application@d94e739490340474f8464f74289a0f48706dd3a3
+      id: run_tests
       with:
-        api-level: 29
-        arch: x86_64
-        force-avd-creation: false
-        emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
-        disable-animations: true
-        script: |
-          adb devices
-          adb logcat -c
-          adb logcat -v time > "${{ github.workspace }}/logcat.txt" &
-
-          adb install ${{ github.workspace }}/Tests/Tests.Android/bin/Release/io.realm.xamarintests-Signed.apk
-          adb shell pm grant io.realm.xamarintests android.permission.READ_EXTERNAL_STORAGE
-          adb shell pm grant io.realm.xamarintests android.permission.WRITE_EXTERNAL_STORAGE
-
-          echo " --baasurl=${{ secrets.BaseUrl }} --baascluster=${{ inputs.clusterName }} --baasapikey=${{ secrets.AtlasPublicKey}} --baasprivateapikey=${{ secrets.AtlasPrivateKey}} --baasprojectid=${{ secrets.AtlasProjectId }}" > ${{ github.workspace }}/testargs.txt
-          adb push ${{ github.workspace }}/testargs.txt /storage/emulated/0/RealmTests/testargs.txt
-
-          adb shell am instrument -w -r io.realm.xamarintests/.TestRunner
-          adb pull /storage/emulated/0/RealmTests/TestResults.Android.xml ${{ github.workspace }}/TestResults.Android.xml
-          adb shell rm /sdcard/Realmtests/TestResults.Android.xml
-    - name: Output logcat
+        project_arn: ${{ secrets.DEVICEFARM_PROJECT_ARN }}
+        device_pool_arn: ${{ secrets.DEVICEFARM_ANDROID_POOL_ARN }}
+        app_file: ${{ github.workspace }}/Tests/Tests.Android/bin/Release/io.realm.xamarintests-Signed.apk
+        app_type: ANDROID_APP
+        test_type: APPIUM_PYTHON
+        test_package_file: https://github.com/realm/aws-devicefarm-sample-data/releases/download/0.3/sample_env_python3.zip
+        test_package_type: APPIUM_PYTHON_TEST_PACKAGE
+        test_spec_file: test_spec.yaml
+        test_spec_type: APPIUM_PYTHON_TEST_SPEC
+        remote_src: true
```

Comment:
These don't seem like something that should be configurable. It's fine to leave them like this for now, but I'd rather set them as defaults in the action and not have us specify them every time.

Reply:
I'll wait until I have a working version for iOS as well, or until we need to integrate this into other SDKs. It's not unlikely that other SDKs will use a different test package or spec. But I agree, this can use a refactoring on a second pass.